src/Services/AppointmentService.php

```diff
@@ -84,17 +92,41 @@ public function getUuidFields(): array
         return ['puuid', 'pce_aid_uuid', 'pc_uuid', 'facility_uuid', 'billing_location_uuid' ];
     }
 
+    /**
+     * Validates the data for updating an appointment.
+     *
+     * @param array $appointment An array containing the appointment data to be validated.
+     * @return ValidationResult Returns the result of the validation operation.
+     */
+    public function update_validate(array $appointment)
+    {
+        $validator = new Validator();
+
+        $validator->optional('pc_title')->string();
+        $validator->optional('pc_room')->string();
+        $validator->optional('pc_eventDate')->datetime('Y-m-d');
+        $validator->optional('pc_hometext')->string();
+        $validator->optional('pc_startTime')->length(5);
+        $validator->optional('pc_endTime')->length(5);
+        $validator->optional('pc_duration')->numeric();
+        $validator->optional('pc_apptstatus')->string();
+
+        return $validator->validate($appointment);
+    }
+
     public function validate($appointment)
```

Comment:
Validation.php
class Validation {
public function validate(array $appointment) {...}

}

And later you can add this class on the constructor method of this class [AppointmentService]